engine/executor/server.go

```diff
@@ -751,13 +721,18 @@ func (s *Server) bgUpdateServerMasterClients(ctx context.Context) error {
 		select {
 		case <-ctx.Done():
 			return perrors.Trace(ctx.Err())
-		case info, ok := <-s.cliUpdateCh:
-			if !ok {
-				// s.cliUpdateCh is closed.
-				return nil
+		case <-time.After(defaultDiscoveryAutoSyncInterval):
```

Comment:
Please consider using a `Ticker` unless it's absolutely necessary to use `time.After`. The reason is that `time.After` can leak timers.

Reply:
`Ticker` will ignore the request latency. Is it possible to leak the timer every time it is created, or just when clicking "<-ctx.Done()".